### PR TITLE
Add remove complete class connect

### PR DESCRIPTION
**Pull request by car12lin12**
resolves stud239
### User Flow :
- user is logged in.
- user selects profile from navigation menu.
- user selects gear icon on top right to edit profile.
- buttons to add, remove, completed classes are displayed in their respective classes taken and classes enrolled sections.
- user selects a button to add, remove or complete a class then a modal/popup appears requesting for text input of class name (eg engr392) that is case insensitive 
- user inputs engr392 and selects submit, then these changes are seen immediately on client side, also completed on backend.
- alternate: user cancels instead of submit and the modal is closed, no changes to classes are made.
- user selects gear icon to exit edit mode and the buttons to add, remove, complete classes are hidden again. Any changes they had made to classes are seen in the classes taken or classes enrolled sections.

### Notable files for add, remove, complete classes:
- UserProfile.js
- AddClassesTakenModal.js
- AddCurrentClassModal.js
- RemoveClassesTakenModal.js
- RemoveCurrentClassmodal.js
- CompleteClassModal.js

I also pass around the bearer token (I know we wanted to use async storage but not sure when that will happen or if it will) so we dont have to always change the bearer token. Both passing and having it static is for testing purposes, static is good for general development to avoid login, then passing it around will be useful when system testing so I'd like to keep both options available.

### Also Completed:
- extract claims from jwt token to get userID from login and pass to handleLogin so we use dynamic userID 
- connect profile/api/reviews-by/:reviewer that gets reviews for the user profile
- a lot of files changed partly bc of my linter changing format

#### Tested with these credentials:
{ "email": "michaelscott@gmail.com", "password": "password"}
OR
{  "email": "dwightschrute@gmail.com", "password": "password"}

### CODE REVIEW

File: src/components/LoginContent.js [hunk 1]
```diff
@@ -29,6 +33,8 @@ class LoginContent extends Component {
             hideConfirmationPassword: true,
             isPasswordConfirmed: false,
             isRegisterReady: false,
+            userID: null,
+            token: null
```
Comment (nuzurie) [inline src/components/LoginContent.js]: Can you make these '' (empty strings) instead of null?

**Review by nuzurie — APPROVED** [review]
Looks great to me! Very thorough and impressive work!

File: src/components/AddClassesTakenModal.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    this.handleSubmit()
+
+                  }}
+                >
+                  <Text style={styles.textStyle}>Submit</Text>
+                </Pressable>
+                <View style={styles.buttonSpace} />
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
+                  onPress={() => this.setModalVisible(!modalVisible)}
```
Comment (yasfahm) [inline src/components/AddClassesTakenModal.js:90]: can you change the background colour to theme.COLOR_RED?

File: src/components/AddClassesTakenModal.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+              <Text style={styles.modalText}>Add A Class Taken</Text>
+
+              <TextInput
+                style={styles.input}
+                onChangeText={text => this.setState({ text })}
+              />
+
+              <View style={styles.modalButtons}>
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
```
Comment (yasfahm) [inline src/components/AddClassesTakenModal.js]: can you change the background colour to theme.COLOR_GREEN?

File: src/components/AddCurrentClassModal.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+              <Text style={styles.modalText}>Add A Current Class</Text>
+
+              <TextInput
+                style={styles.input}
+                onChangeText={text => this.setState({ text })}
+              />
+
+              <View style={styles.modalButtons}>
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
```
Comment (yasfahm) [inline src/components/AddCurrentClassModal.js]: can you change the background colour to theme.COLOR_GREEN?

File: src/components/AddCurrentClassModal.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                    this.handleSubmit()
+
+                  }}
+                >
+                  <Text style={styles.textStyle}>Submit</Text>
+                </Pressable>
+                <View style={styles.buttonSpace} />
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
+                  onPress={() => this.setModalVisible(!modalVisible)}
```
Comment (yasfahm) [inline src/components/AddCurrentClassModal.js:90]: can you change the background colour to theme.COLOR_RED?

File: src/components/CompleteClassModal.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+              <Text style={styles.modalText}>Complete A Current Class</Text>
+
+              <TextInput
+                style={styles.input}
+                onChangeText={text => this.setState({ text })}
+              />
+
+              <View style={styles.modalButtons}>
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
```
Comment (yasfahm) [inline src/components/CompleteClassModal.js]: can you change the background colour to theme.COLOR_GREEN?

File: src/components/CompleteClassModal.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+                    this.setModalVisible(!modalVisible)
+                    this.handleSubmit()
+
+                  }}
+                >
+                  <Text style={styles.textStyle}>Submit</Text>
+                </Pressable>
+                <View style={styles.buttonSpace} />
+                <Pressable
+                  style={[styles.button, styles.buttonClose]}
```
Comment (yasfahm) [inline src/components/CompleteClassModal.js]: can you change the background colour to theme.COLOR_RED?

**Review by yasfahm — APPROVED** [review]

**Comment by yasfahm** [comment]
Can you also reduce the width of all Add/Remove/Complete Class buttons?

**Comment by yasfahm** [comment]
It looks much better now!! :partying_face:
I was also thinking maybe we can use a **garbage** for remove, **+ sign** for add and **checkmark** for complete class
Screen Shot 2022-03-19 at 8 36 43 PM
We can wait for @stellakimnguyen 's input on this 😄

**Review by stellakimnguyen — APPROVED** [review]
Looks amazing! I would suggest removing the package-lock before merging

**Pull request merged**